src/main/java/org/fogbowcloud/arrebol/execution/DockerTaskExecutor.java

```diff
@@ -1,146 +1,200 @@
+        Arrays.fill(commandsResults, TaskExecutionResult.UNDETERMINED_RESULT);
+        for (int i = 0; i < commands.length; i++) {
             Command c = commands[i];
             c.setState(CommandState.RUNNING);
             try {
                 Integer exitCode = executeCommand(c);
                 commandsResults[i] = exitCode;
+                c.setExitcode(exitCode);
                 c.setState(CommandState.FINISHED);
             } catch (Throwable t) {
                 c.setState(CommandState.FAILED);
             }
-
-        }
-
-        Integer stopStatus = this.stop();
-        if(stopStatus != 0){
-            LOGGER.error("Exit code from container stop: " + stopStatus);
         }
+        return commandsResults;
+    }
 
+    protected TaskExecutionResult getTaskResult(Command[] commands, int[] commandsResults){
+        TaskExecutionResult taskExecutionResult;
         TaskExecutionResult.RESULT result = TaskExecutionResult.RESULT.SUCCESS;
         for (Command cmd : commands) {
             if (cmd.getState().equals(CommandState.FAILED)) {
                 result = TaskExecutionResult.RESULT.FAILURE;
                 break;
             }
         }
-
-        LOGGER.info("Result of task" + task.getId() + ": " + result.toString());
-
         taskExecutionResult = new TaskExecutionResult(result, commandsResults, commands);
         return taskExecutionResult;
     }
 
-    private Integer start() {
+    protected Integer start(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_RUN + " " + this.containerName + " " + this.imageId
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to start container " + this.containerName);
+            LOGGER.info("Starting DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.start();
+            return new Integer(0);
+        } catch (Exception e) {
+            return new Integer(127);
         }
     }
 
-    private Integer stop() {
+    protected Integer stop(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_STOP + " " + this.containerName
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to stop container " + this.containerName);
+            LOGGER.info("Stopping DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.stop();
+            return new Integer(0);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
why do have this print here?

Reply:
Solved in 3f2467d41957697d2578c4d718c4907e36824057.